shared_buffer_pool.go

```diff
@@ -114,46 +114,33 @@ type simpleSharedBufferChildPool interface {
 
 type bufferPool struct {
 	sync.Pool
+
+	defaultSize int
 }
 
 func (p *bufferPool) Get(size int) []byte {
 	bs := p.Pool.Get().(*[]byte)
 
+	// If default size is 0. It means this pool is fallback pool.
+	// Therefore, we need to make a new one if the requested size is larger than the buffer.
+	if p.defaultSize == 0 {
+		if cap(*bs) < size {
+			*bs = make([]byte, size)
```

Comment:
Did we want to put the old `*bs` back into the pool here?

Reply:
This is an excellent observation that you've made. Thank you! 2678efb1a7844ea090938abdb1d3e0f7288f43b4